### PR TITLE
Added Tcp_check agent

### PR DESCRIPTION
**Pull request by aepod**
See: http://docs.datadoghq.com/integrations/tcpcheck/

This should allow for the tcp_check to be used and it has all the required spec files etc.

### CODE REVIEW

File: manifests/integrations/tcp_check.pp [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+#
+#   timeout
+#       (Optional) - Timeout for the check. Defaults to 10 seconds.
+#
+#   threshold
+#       (Optional) - Used in conjunction with window. An alert will 
+#        trigger if the check fails <threshold> times in <window> attempts.
+#
+#   window
+#       (Optional) - Refer to �threshold�.
```
Comment (truthbk) [inline manifests/integrations/tcp_check.pp]: @aepod the quotes here are not ASCII quotes, this is what's preventing the CI from running your tests. You should probably change them to regular quotes, or just remove them.

File: manifests/integrations/tcp_check.pp [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+#        {
+#          'name'  => 'www.example.com-https',
+#          'host'  => 'www.example.com',
+#          'port'       => '443',
+#        }]
+#     }
+
+
+class datadog_agent::integrations::tcp_check (
+  $name      = undef,
```
Comment (truthbk) [inline manifests/integrations/tcp_check.pp]: You will probably have to change this parameter name to something else or you'll get this error: `The parameter $name redefines a built in parameter in the Host Class Definition`. May I suggest `sitename` or `hostname`?

File: manifests/integrations/tcp_check.pp [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+  $host      = undef,
+  $port      = undef,
```
Comment (truthbk) [inline manifests/integrations/tcp_check.pp]: A required parameter should not have an optional value. I believe `host` and `port` are definitely mandatory. I would probably make `sitename` mandatory as well just to be safe.

File: templates/agent-conf.d/tcp_check.yaml.erb [hunk 4]
```diff
@@ -0,0 +1,31 @@
+init_config:
+
+instances:
+<%- (Array(@_instances)).each do |instance| -%>
+  - name: <%= instance['name'] %>
```
Comment (truthbk) [inline templates/agent-conf.d/tcp_check.yaml.erb]: You will have to update this to `instance['sitename']` or whatever.

File: manifests/integrations/tcp_check.pp (hunk 1 above)
Comment (aepod) [inline manifests/integrations/tcp_check.pp]: Interesting copy/paste/edit from the documentation on the check, oops. those quotes snuck through. I assume this was the "non-utf8" error, which was bugging me! haha I will sort this out, thanks for the heads up.

File: manifests/integrations/tcp_check.pp (hunk 2 above)
Comment (aepod) [inline manifests/integrations/tcp_check.pp]: Makes sense, I sort of knew that when I switched it, but wasn't sure. I will switch it to something else and resubmit.

**Comment by aepod** [comment]
This now tests out, and works. Looks good and should be ready for merging. Thanks for the help on the pull request earlier.

**Comment by truthbk** [comment]
Thanks @aepod! This looks much better now, merging. 🙇

**Pull request merged**